### PR TITLE
Fix ordering in transaction API for PostgreSQL

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

Similar fix to #1516. Only the latest operation for each key and ignore the other operations. This ensures that the ordering of the operations is intact.

## Issue reference

Please reference the issue this PR will close: #1522

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* (NA) Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: state/postgresql/postgresql_test.go [hunk 1]
```diff
@@ -52,10 +53,20 @@ func (m *fakeDBaccess) Get(req *state.GetRequest) (*state.GetResponse, error) {
...
 
 func (m *fakeDBaccess) Delete(req *state.DeleteRequest) error {
+	m.deleteExecuted = true
+
 	return nil
 }
 
 func (m *fakeDBaccess) ExecuteMulti(sets []state.SetRequest, deletes []state.DeleteRequest) error {
+	for _, delete := range deletes {
+		m.Delete(&delete)
```
Comment (Taction) [inline state/postgresql/postgresql_test.go:63]: delete := delete
m.Delete(&delete)

File: state/postgresql/postgresql_test.go [hunk 2]
```diff
@@ -52,10 +53,20 @@ func (m *fakeDBaccess) Get(req *state.GetRequest) (*state.GetResponse, error) {
...
+	}
+
+	for _, set := range sets {
+		m.Set(&set)
```
Comment (Taction) [inline state/postgresql/postgresql_test.go:67]: set := set
m.Set(&set)

**Comment by Taction** [comment]
@shubham1172 Please fix linter issues. Search for "Check failure" here: https://github.com/dapr/components-contrib/pull/1524/files. It'll show you exactly which files need what formatter.

**Comment by shubham1172** [comment]
@Taction thanks for your reviews, I have marked this as WIP for now, as I need to change the core logic as discussed in #1518

**Review by artursouza — APPROVED** [review]

**Pull request merged**